framework/include/materials/MaterialData.h

```diff
@@ -161,6 +161,32 @@ class MaterialData
    */
   bool isOnlyResizeIfSmaller() const { return _resize_only_if_smaller; };
 
+  /**
+   * Copy all property values common between this material data and 'other'
+   * from other into this object.
+   */
+  void copyPropsFrom(MaterialData & other)
+  {
+    for (size_t i = 0; i < other._props.size(); i++)
+    {
+      if (_props.size() <= i)
+        break;
+      auto src = other._props[i];
```

Comment:
Well - `auto&` is worse because then we have a reference to a pointer - which is slower and not exactly the semantics we need.  Also - you have the assurance that I wrote the code - and I know what I am doing - I never make mistakes like that :-P

Reply:
I definitely agree that `auto &` would be worse. I would advocate for `MaterialProperties *` or whatever the object is...see with `auto` I really have no idea what it is. I  think I can call on @aeslaughter for support on this one

Reply:
It's also nice as a developer if I'm using `etags` or the equivalent for other editors, if the type is explicitly stated *and* I don't know anything about that type, then I can click `M-.` and then I am instantly directed to the class definition. Can't do that with `auto`. FTR I am definitely not saying you should change this @rwcarlsen since our developer documentation explicitly states that you should do what you are doing